### PR TITLE
feat(twig-paths): better resolve of twig component path with twig.yaml config

### PR DESCRIPTION
**Pull request by Taloud**
We are now using the paths in twig.yaml config to resolve the twig_components paths.

Fix of https://github.com/sensiolabs/StorybookBundle/issues/31

~~TODO: update the tests suite~~

### CODE REVIEW

File: storybook/package.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+    "version": "0.0.0",
```
Comment (Taloud) [inline storybook/package.json:2]: **Note**: Needed older versions of npm / yarn. Can't build without it.

**Comment by Taloud** [comment]
Just a quick follow-up on this 👼. I’m available to discuss it if necessary.

**Comment by Taloud** [comment]
Hello, I don’t really understand why I’m getting PHP CS Fixer issues when I’m not touching a single PHP file. I can run the fixer command, but it doesn’t seem related to me, right?

I will look into the sandbox because I don’t know much about it either, and I will try to fix this. Thanks

**Comment by squrious** [comment]
Hello, first of alls sorry for the delay, didn't get free time to work on the projet during past weeks. 

You can safely ignore the PHP CS fixer errors. The CI always uses the latest version/rules, and sometimes it breaks, but I'll fix that in another PR. 

For the sandbox, yes there is a problem actually. My guess is it is about how the configurations are processed in `framework-preset.ts`. If the Symfony project doesn't configure paths explicitely, then you have no paths to use in the resolver. A fix could be to append the default template directory if no one is provided:

```ts
    // ...
    const twigPaths: string[] = Object.keys(twigConfig.paths).map((key) => `${projectDir}/${key}/`);
     
     // Fix:
     if(twigPaths.length === 0) {
         twigPaths.push(`${projectDir}/templates`);
     }
    // ...
```

The global implementation in you PR looks great to me, thank you for your work! I'll make a deeper review soon ;)

**Comment by Taloud** [comment]
Hello, no problem, and thanks for the review! 

I’ve pushed the fix. I don’t really know how to run the sandbox test, but I’ll look into it more if needed.

File: storybook/src/server/lib/symfony.ts [hunk 2]
```diff
@@ -119,34 +119,53 @@ export const getTwigComponentConfiguration = async () => {
...
+};
+
+export type TwigConfiguration = {
+    paths: string[];
+};
+
 /**
  * Attempt to resolve the Twig template path containing sources for the given TwigComponent.
  */
 export function resolveTwigComponentFile(componentName: string, config: TwigComponentConfiguration) {
```
Comment (squrious) [inline storybook/src/server/lib/symfony.ts:147]: Can you remove this function? It's not used anymore and has been replace by `TwigComponentResolver`

File: storybook/src/server/framework-preset.ts [hunk 3]
```diff
@@ -65,7 +80,10 @@ export const webpack: StorybookConfig['webpack'] = async (config, options) => {
                           projectDir: symfonyOptions.projectDir,
                           additionalWatchPaths: symfonyOptions.additionalWatchPaths,
                       }),
-                TwigLoaderPlugin.webpack({ twigComponentConfiguration: symfonyOptions.twigComponent }),
+                TwigLoaderPlugin.webpack({
+                    twigComponentConfiguration: symfonyOptions.twigComponent,
+                    twigConfiguration: symfonyOptions.twig,
```
Comment (squrious) [inline storybook/src/server/framework-preset.ts:85]: It looks like you don't need to pass the twig configuration here, as you built all the paths in `getBuildOptions`

**Review by squrious — CHANGES_REQUESTED** [review]

File: storybook/src/server/framework-preset.ts (hunk 3 above)
Comment (Taloud) [reply storybook/src/server/framework-preset.ts]: indeed everything works without it. Thanks and done

File: storybook/src/server/lib/symfony.ts (hunk 2 above)
Comment (Taloud) [reply storybook/src/server/lib/symfony.ts]: done

**Review by squrious — APPROVED** [review]

**Comment by squrious** [comment]
Thank you @Taloud!

**Pull request merged**